plugins/mesh_tools.js

```diff
@@ -1715,6 +1806,421 @@
+									let vinc = (vmax - vmin)/vsteps;
+									
+									let uRange = usteps+1;
+									let vRange = vsteps+1;
+									if (s.uWrap) uRange -= 1;
+									if (s.vWrap) vRange -= 1;
+
+									let vertices = [];
+									for (let j = 0; j < vRange; j++) {
+										let v = uinc + (j*vinc);;
+										for (let i = 0; i < uRange; i++) {
+											let u = vinc + (i*uinc);
+											
+											mtMolangParser.global_variables = {'p.u': u,'p.v': v};
+
+											let x = mtMolangParser.parse(out.x);
+											let y = mtMolangParser.parse(out.y);
+											let z = mtMolangParser.parse(out.z);
+											vertices.push(mesh.addVertices([x,y,z])[0]);
+										}
+									}
+									for (let y = 0; y < vsteps; y++) {
+										for (let x = 0; x < usteps; x++) {
+											yNext = y+1;
+											xNext = x+1;
+											
+											if (out.vWrap && yNext >= vRange) yNext = 0;
+											if (out.uWrap && xNext >= uRange) xNext = 0;
+
+											let face = new MeshFace(mesh, {vertices: [
+												vertices[(yNext * uRange) + xNext],
+												vertices[(yNext * uRange) + x],
+												vertices[	 (y * uRange) + x],
+												vertices[    (y * uRange) + xNext],											
+											]});
+											mesh.addFaces(face);
+										}
+									}
+									// fills end caps
+									if (out.vClose && out.uWrap && !out.vWrap) {
+										for (let x = 1; x < usteps-1; x++) {
+											let face1 = new MeshFace(mesh, {vertices: [
+												vertices[usteps - 1],
+												vertices[usteps - 1 - x],
+												vertices[usteps - 2 - x],							
+											]});
+											let face2 = new MeshFace(mesh, {vertices: [
+												vertices[vsteps * uRange],
+												vertices[vsteps * uRange + x],
+												vertices[vsteps * uRange + x + 1],							
+											]});
+											mesh.addFaces(face1, face2);
+										}
+									}
+									mesh.init();
+									elements.push(mesh);
+									mesh.select();
+									Undo.finishEdit('Generate Mesh');
+								}
+								runEdit(out);
+							}
+						})
+						dial.show();
+						if(msettings != null) {
+							dial.setFormValues(msettings);
+						}
+
+						let saveBtn = $(`button:contains("Save Settings To Memory")`);
+						saveBtn.off("click");
+						saveBtn.on("click", function() {
+							let mmsettings = dial.getFormResult();
+							localStorage.setItem("mt_xyzSettings", JSON.stringify(mmsettings));
+						});
+					}
+				})
+			);
+			let quickPrimitives = [];
+			let quickPrimitivesI = "offline_bolt";
+			quickPrimitives.push(
+				new Action("polyhedron", {
+					name: "Polyhedron",
+					icon: quickPrimitivesI,
+					click(){
+						new Dialog ({
+							title: "Quick Primtive [ Polyhedron ]",
+							form: {
+								select: {label: "Hedron", type:"select", options: {Icosahedron:"Icosahedron", Dodecahedron:"Dodecahedron",Octahedron:"Octahedron", Tetrahedron:"Tetrahedron"}},
+								radius: {label: "Radius", value: 8},

```

Comment:
These two lines should be of type "number", especially because they are used in eval. Using plain strings from user inputs would be a security risk.

Reply:
I'd generally avoid eval if possible. In this case, you can do the same thing without it.
```
new THREE[out.select + 'BufferGeometry'](s.radius, s.detail-1)
```